### PR TITLE
Verify improved errors for name lookup

### LINKED ISSUE #7493: Error message change for type not found is not correct
```
using A = List<Bar>;
```
Reports in Roslyn:
```
Error CS0246: The type or namespace name 'List<Bar>' could not be found (are you missing a using directive or an assembly reference?)
```
In legacy compiler:
```
Error CS0246: The type or namespace name 'List' could not be found (are you missing a using directive or an assembly reference?)
```
To me, this is possibly confusing. In this case it is `List<T>` that could not be found. Not `List<Bar>`. And especially when going into some silly variations, eg `List<"moo">`.

I can understand it was a possible enhancement, but it would be better to say `no generic type 'List' taking 1 argument`.

Unfortunately, the Git history does not go back far enough to say why this change was done. 

The point of interest is the `whereText` parameter in `Binder.NotFound(...)`. See: https://github.com/dotnet/roslyn/blob/master/src%2FCompilers%2FCSharp%2FPortable%2FBinder%2FBinder_Symbols.cs#L1742

Related: #7387

### PR DESCRIPTION
**Pull request by gafter**
Fixes #7493 
@dotnet/roslyn-compiler Please review. This is just the addition of one small test.

### CODE REVIEW

**Comment by cston** [comment]
LGTM

**Comment by TyOverby** [comment]
LGTM

File: src/Compilers/CSharp/Test/Semantic/Semantics/LookupTests.cs [hunk 1]
```diff
@@ -1523,6 +1523,20 @@ void M()
...
         }
 
+        [WorkItem(7493, "https://github.com/dotnet/roslyn/issues/7493")]
+        [Fact]
+        public void GenericNameLookup()
+        {
+            var source = @"using A = List<int>;";
+            var compilation = CreateCompilationWithMscorlib(source).VerifyDiagnostics(
+                // (1,11): error CS0246: The type or namespace name 'List<>' could not be found (are you missing a using directive or an assembly reference?)
+                // using A = List<Bar>;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LookupTests.cs]: This comment doesn't match the source.

File: src/Compilers/CSharp/Test/Semantic/Semantics/LookupTests.cs [hunk 2]
```diff
@@ -1523,6 +1523,20 @@ void M()
...
+                Diagnostic(ErrorCode.ERR_SingleTypeNameNotFound, "List<int>").WithArguments("List<>").WithLocation(1, 11),
+                // (1,1): hidden CS8019: Unnecessary using directive.
+                // using A = List<Bar>;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/LookupTests.cs]: Comment doesn't match the source.

**Comment by AlekseyTs** [comment]
LGTM, modulo the comments adjustment.

**Pull request merged**